pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,365 @@
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), lvol.Name),
+            Body: BlockingMessage(usage)
+        });
+        return;
+    }
+
+    let grow_size;
+    let size_fields = [];
+    if (!to_fit) {
+        size_fields = [
+            SizeSlider("size", _("Size"),
+                       {
+                           value: lvol.Size,
+                           min: lvol.Size,
+                           max: (pool ? pool.Size * 3 : lvol.Size + vgroup.FreeSize),
+                           allow_infinite: !!pool,
+                           round: vgroup.ExtentSize
+                       })
+        ];
+    } else {
+        grow_size = block.Size;
+    }
+
+    let recovered_passphrase;
+    let passphrase_fields = [];
+    if (block && block.IdType == "crypto_LUKS" && block.IdVersion == 2)
+        passphrase_fields = existing_passphrase_fields(_("Resizing an encrypted filesystem requires unlocking the disk. Please provide a current disk passphrase."));
+
+    if (!usage.Teardown && size_fields.length + passphrase_fields.length === 0) {
+        return lvol_and_fsys_resize(client, lvol, grow_size, info.grow_needs_unmount, null);
+    }
+
+    const dlg = dialog_open({
+        Title: _("Grow logical volume"),
+        Teardown: TeardownMessage(usage),
+        Fields: size_fields.concat(passphrase_fields),
+        Action: {
+            Title: _("Grow"),
+            action: function (vals) {
+                return teardown_active_usage(client, usage)
+                        .then(function () {
+                            return (lvol_and_fsys_resize(client, lvol,
+                                                         to_fit ? grow_size : vals.size,
+                                                         info.grow_needs_unmount,
+                                                         vals.passphrase || recovered_passphrase)
+                                    .catch(request_passphrase_on_error_handler(dlg, vals, recovered_passphrase, block)));
+                        });
+            }
+        },
+        Inits: [
+            init_active_usage_processes(client, usage),
+            passphrase_fields.length
+                ? init_existing_passphrase(block, false, pp => { recovered_passphrase = pp })
+                : null
+        ]
+    });
+}
+
+export function shrink_dialog(client, lvol, info, to_fit) {
+    const block = client.lvols_block[lvol.path];
+    const vgroup = client.vgroups[lvol.VolumeGroup];
+
+    const usage = get_active_usage(client, block && !to_fit && info.shrink_needs_unmount ? block.path : null,
+                                   _("shrink"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), lvol.Name),
+            Body: BlockingMessage(usage)
+        });
+        return;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19400-20230927-121722-6ccb869e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)